common/src/stack/command/stack/commands/remove/storage/partition/__init__.py

```diff
@@ -46,49 +50,25 @@ class Command(stack.commands.remove.command,
 	"""
 
 	def run(self, params, args):
-		scope = None
+		(scope, device, mountpoint) = self.fillParams([
+			('scope', 'global'), ('device', None), ('mountpoint', None)])
 		oses = []
 		appliances = []
 		hosts = []
+		name = None
 
-		if len(args) == 0:
-			scope = 'global'
-		elif len(args) == 1:
-			try:
-				oses = self.getOSNames(args)
-			except:
-				oses = []
-
-			try:
-				appliances = self.getApplianceNames(args)
-			except:
-				appliances = []
-
-			try:
-				hosts = self.getHostnames(args)
-			except:
-				hosts = []
-		else:
-			raise ArgError(self, 'scope', 'must be unique or missing')
-
-		if not scope:
-			if args[0] in oses:
-				scope = 'os'
-			elif args[0] in appliances:
-				scope = 'appliance'
-			elif args[0] in hosts:
-				scope = 'host'
-
-		if not scope:
-			raise ParamValue(self, 'scope', 'valid os, appliance name or host name')
+		if scope == "os":
+			oses = self.getOSNames(args)
+		elif scope == "appliance":
+			appliances = self.getApplianceNames(args)
+		elif scope == "host":
+			hosts = self.getHostnames(args)
 
-		if scope == 'global':
-			name = None
-		else:
-			name = args[0]
+		if scope != 'global' and len(args) < 1:
```

Comment:
Since we're only going to support one entity at a time, I'd suggest putting in a check and raising an exception if `len(args) != 1`